File: branches/Medusa/Source/Fantasy.Jobs/RegisterIpcChannelCommand.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Runtime.Remoting.Channels.Ipc;
using System.Reflection;
using System.IO;
using System.Diagnostics;
using System.Runtime.Remoting.Channels;
using System.Collections;
using System.Runtime.Serialization.Formatters;

namespace Fantasy.Jobs
{
    public class RegisterIpcChannelCommand : ICommand
    {
        #region ICommand Members

        public object Execute(object args)
        {
            IDictionary props = new Hashtable();
            props["secure"] = true;
            props["tokenImpersonationLevel"] = "impersonation";
            props["authorizedGroup"] = "Everyone";
            props["portName"] = Process.GetCurrentProcess().ProcessName + ":" + Process.GetCurrentProcess().Id.ToString();
            IpcChannel ipc = new IpcChannel(props, new BinaryClientFormatterSinkProvider(), new BinaryServerFormatterSinkProvider() { TypeFilterLevel = TypeFilterLevel.Full });
            ChannelServices.RegisterChannel(ipc, true);
            return null;

        }

        #endregion
    }
}

File: Fantasy.Jobs/src/fantasy/jobs/RegisterIpcChannelCommand.java
package fantasy.jobs;

public class RegisterIpcChannelCommand implements ICommand
{
//C# TO JAVA CONVERTER TODO TASK: There is no preprocessor in Java:
		///#region ICommand Members

	public final Object Execute(Object args)
	{
		java.util.Map props = new java.util.Hashtable();
		props.put("secure", true);
		props.put("tokenImpersonationLevel", "impersonation");
		props.put("authorizedGroup", "Everyone");
		props.put("portName", Process.GetCurrentProcess().ProcessName + ":" + Process.GetCurrentProcess().Id.toString());
		BinaryServerFormatterSinkProvider tempVar = new BinaryServerFormatterSinkProvider();
		tempVar.TypeFilterLevel = TypeFilterLevel.Full;
		IpcChannel ipc = new IpcChannel(props, new BinaryClientFormatterSinkProvider(), tempVar);
		ChannelServices.RegisterChannel(ipc, true);
		return null;

	}

//C# TO JAVA CONVERTER TODO TASK: There is no preprocessor in Java:
		///#endregion
}